In [56]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split 
from sklearn import preprocessing
from sklearn.naive_bayes import *
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix



In [18]:
data=pd.read_csv("C:/Users/user/Desktop/lets upgrade/day24/train.csv")
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [48]:
data.isnull().sum()
d3=data.drop(['Cabin','Name','PassengerId','Ticket'],axis=1)

In [49]:
print(d3.isnull().sum())
print(d3.columns)


Survived    0
Pclass      0
Sex         0
Age         0
SibSp       0
Parch       0
Fare        0
Embarked    0
dtype: int64
Index(['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare',
       'Embarked'],
      dtype='object')


In [50]:
le=preprocessing.LabelEncoder()
d3["Sex"]=le.fit_transform(d3['Sex'])
d3['Embarked']=le.fit_transform(d3['Embarked'])
d3.head(5)

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

In [55]:
y=d3['Survived']
X=d3[['Pclass','Sex',"Age",'SibSp','Parch','Fare','Embarked']]
print(X)
print(y)

     Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         3    1  22.0      1      0   7.2500         2
1         1    0  38.0      1      0  71.2833         0
2         3    0  26.0      0      0   7.9250         2
3         1    0  35.0      1      0  53.1000         2
4         3    1  35.0      0      0   8.0500         2
..      ...  ...   ...    ...    ...      ...       ...
884       2    1  27.0      0      0  13.0000         2
885       1    0  19.0      0      0  30.0000         2
886       3    0  45.0      1      2  23.4500         2
887       1    1  26.0      0      0  30.0000         0
888       3    1  32.0      0      0   7.7500         1

[889 rows x 7 columns]
0      0
1      1
2      1
3      1
4      0
      ..
884    0
885    1
886    0
887    1
888    0
Name: Survived, Length: 889, dtype: int64


In [58]:
x_train,x_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0) # randomly classfying without any condition 

In [84]:
print(x_test.shape)

(267, 7)


In [77]:
clf=BernoulliNB()
clf.fit(x_train,y_train)
ypred=clf.predict(x_test)


In [82]:
print(accuracy_score(y_test,ypred,normalize=1))
print(confusion_matrix(y_test,ypred))

0.7715355805243446
[[131  26]
 [ 35  75]]


In [90]:
d3.columns

Index(['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare',
       'Embarked'],
      dtype='object')

In [92]:
X1=d3[['Survived','Sex','Age','SibSp','Parch','Fare','Embarked']]
y1=d3['Pclass']

In [94]:
x_train1,x_test1,y_train1,y_test1=train_test_split(X1,y1,test_size=0.3,random_state=0)
clf.fit(x_train1,y_train1)
ypred2=clf.predict(x_test1)

In [95]:
print(accuracy_score(y_test1,ypred2,normalize=1))
print(confusion_matrix(y_test1,ypred2))

0.5917602996254682
[[ 30   8  32]
 [  9  10  30]
 [ 24   6 118]]


In [119]:
X2=d3[['Survived','Pclass','Age','SibSp','Parch','Fare','Embarked']]
y2=d3['Sex']

In [120]:
x_train2,x_test2,y_train2,y_test2=train_test_split(X2,y2,test_size=0.3,random_state=0)
clf.fit(x_train2,y_train2)
ypred3=clf.predict(x_test2)

In [125]:
print(accuracy_score(y_test2,ypred3,normalize=1))
print(confusion_matrix(y_test2,ypred3))

0.7453183520599251
[[ 49  49]
 [ 19 150]]


In [131]:
X3=d3[['Survived','Pclass','Age','Sex','Parch','Fare','Embarked']]
y3=d3['SibSp']

In [134]:
x_train3,x_test3,y_train3,y_test3=train_test_split(X3,y3,test_size=0.3,random_state=0)
clf.fit(x_train3,y_train3)
ypred4=clf.predict(x_test3)

In [135]:
print(accuracy_score(y_test3,ypred4,normalize=1))
print(confusion_matrix(y_test3,ypred4))

0.6891385767790262
[[162  20   0   0   0   0   0]
 [ 43  22   0   0   0   0   0]
 [  6   2   0   0   0   0   0]
 [  5   2   0   0   0   0   0]
 [  2   0   0   0   0   0   0]
 [  1   0   0   0   0   0   0]
 [  2   0   0   0   0   0   0]]


In [142]:
X4=d3[['Survived','Pclass','Age','Sex','SibSp','Fare','Embarked']]
y4=d3['Parch']

In [143]:
x_train4,x_test4,y_train4,y_test4=train_test_split(X4,y4,test_size=0.3,random_state=0)
clf.fit(x_train4,y_train4)
ypred5=clf.predict(x_test4)

In [144]:
print(accuracy_score(y_test4,ypred5,normalize=1))
print(confusion_matrix(y_test4,ypred5))

0.7153558052434457
[[182  16   0   0   0]
 [ 31   9   0   0   0]
 [ 24   3   0   0   0]
 [  1   0   0   0   0]
 [  1   0   0   0   0]]


In [148]:
X5=d3[['Survived','Pclass','Age','Sex','SibSp','Fare','Parch']]
y5=d3['Embarked']

In [150]:
x_train5,x_test5,y_train5,y_test5=train_test_split(X5,y5,test_size=0.3,random_state=0)
clf.fit(x_train5,y_train5)
ypred6=clf.predict(x_test5)

In [151]:
print(accuracy_score(y_test5,ypred6,normalize=1))
print(confusion_matrix(y_test5,ypred6))

0.7340823970037453
[[  0   0  49]
 [  0   0  22]
 [  0   0 196]]


## Navie Bayes classification

1.

IDV :  'Pclass','Sex',"Age",'SibSp','Parch','Fare','Embarked'      
DV  :   'Survived'         
model_score :  77.15 %          
incorrect prediction  :  22.84 %        

 
 [[131  26 ]
 
 [ 35  75]]                   


2.

IDV :  'Survived','Sex',"Age",'SibSp','Parch','Fare','Embarked'        
DV  :   'Pclass'         
model_score : 59.17 %          
incorrect prediction  : 40.82 %           

[[ 30   8  32]
 [  9  10  30]
 [ 24   6 118]]

3.

IDV :  'Survived','Pclass',"Age",'SibSp','Parch','Fare','Embarked'         
DV  :   'Sex '         
model_score : 74.53%           
incorrect  prediction : 25.46%           

[[ 49  49]
 [ 19 150]]

4.

IDV :  'Survived','Pclass',"Age",'Sex','Parch','Fare','Embarked'           
DV  :   'SibSp '        
model_score : 68.91 %            
incorrect prediction  : 30.33 %            

[[162  20   0   0   0   0   0]
 [ 43  22   0   0   0   0   0]
 [  6   2   0   0   0   0   0]
 [  5   2   0   0   0   0   0]
 [  2   0   0   0   0   0   0]
 [  1   0   0   0   0   0   0]
 [  2   0   0   0   0   0   0]]

5.

IDV :  'Survived','Pclass',"Age",'Sex','SibSp','Fare','Embarked'           
DV  :   ' Parch'           
model_score : 71.53 %           
incorrect  prediction : 27.34 %           

[[182  16   0   0   0]
 [ 31   9   0   0   0]
 [ 24   3   0   0   0]
 [  1   0   0   0   0]
 [  1   0   0   0   0]]

6.

IDV :  'Survived','Pclass',"Age",'Sex','SibSp','Fare',' Parch'           
DV  :   'Embarked'            
model_score :  73.40 %            
incorrect  prediction : 26.59 %             

[[  0   0  49]
 [  0   0  22]
 [  0   0 196]]

